tree/ntuple/v7/test/ntuple_zip.cxx

```diff
@@ -84,3 +84,23 @@ TEST(RNTupleZip, Large)
    decompressor.Unzip(zipBuffer.get(), szZip, N, unzipBuffer.get());
    EXPECT_EQ(data, std::string(unzipBuffer.get(), N));
 }
+
+TEST(RNTupleZip, LargeWithOutputBuffer)
+{
+   constexpr unsigned int N = kMAXZIPBUF + 32;
+   auto zipBuffer = std::make_unique<unsigned char[]>(N);
+   auto unzipBuffer = std::make_unique<char[]>(N);
+   std::string data(N, 'x');
+
+   RNTupleCompressor compressor;
+   RNTupleDecompressor decompressor;
+
+   /// Trailing byte cannot be compressed, entire buffer returns uncompressed
```

Comment:
Parallel question not strictly related to this PR, @jblomer: GitHub doesn't allow me to comment on [RNTupleZip.hxx:149](https://github.com/root-project/root/pull/13202/files#diff-3bcdde6d9c4086f05ff00301119bd5da265584364a7d1454c648b04693766bc8R149), but I wonder whether we should stop at the first uncompressible block -- e.g., in this case here, the overall result is still smaller than the original, given that the previous blocks compressed well.

Reply:
This may be true but then we need to compress all the way through and compare the final result.  The blocks are 16MB, so likely if one is uncompressible, the others are, too.